Makefile

```diff
@@ -1,7 +1,23 @@
-CPPFLAGS += -I /usr/lib/kframework/include -I dummy-version -I plugin -I plugin-c -I libff/build/install/include -I deps/cpp-httplib
-CXX=clang++-8
+# set default install prefix if not set
+PREFIX ?= $(CURDIR)/build
 
-.PHONY: build
+K_RELEASE ?= $(dir $(shell which kompile))..
+
+# set OS specific defaults
+ifeq ($(shell uname -s),Darwin)
+# 1. OSX doesn't have /proc/ filesystem
+# 2. fix cmake openssl detection for brew
+LIBFF_CMAKE_FLAGS += -DWITH_PROCPS=OFF \
+                     -DOPENSSL_ROOT_DIR=/usr/local/opt/$(shell brew desc openssl | cut -f1 -d:)
+else
+# llvm-backend code doesn't play nice with g++
+export CXX := $(if $(findstring default, $(origin CXX)), clang++, $(CXX))
+endif
+
+INCLUDES := -I $(K_RELEASE)/include/kllvm -I $(PREFIX)/include -I dummy-version -I plugin -I plugin-c -I deps/cpp-httplib
+CPPFLAGS += --std=c++14 $(INCLUDES)
```

Comment:
I'm not sure about putting `-I dummy-version` in the `INCLUDES`, because how then can Firefly override this version.

But actually, I think the build system for firefly is just using the dummy version anyway, so this can be fixed in a later PR.